### PR TITLE
Add search as you type

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

File: invenio_records_rest/views.py [hunk 1]
```diff
@@ -941,6 +953,60 @@ def check_etag(self, etag, weak=False):
...
+    def __init__(self, search_as_you_type, search_class=None, **kwargs):
+        """Constructor."""
+        self.search_as_you_type = search_as_you_type
+        self.search_as_you_type_fields = set(self.search_as_you_type.keys())
+        self.search_class = search_class
+
+    def get(self, **kwargs):
+        """Get results from `search_as_you_type`."""
+        request_fields = set(request.values.keys())
+        key = self.search_as_you_type_fields & request_fields
```
Comment (MJedr) [inline invenio_records_rest/views.py]: I would change naming there cause it's not clear to me (is the size of it always 1? if not, why are you taking only one key from it?)

File: tests/test_views_search_as_you_type.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        # Valid simple completion search_as_you_type with size
+        res = client.get(
+            url_for("invenio_records_rest.recid_search_as_you_type"),
+            query_string={"text_with_size": "Back", "size": 2},
+        )
+        data = json.loads(res.get_data(as_text=True))
+        assert len(data["text_with_size"][0]["options"]) == 2
+        assert is_option(exp1, data["text_with_size"][0]["options"])
+
+        # Missing missing and invalid search_as_you_type
```
Comment (MJedr) [inline tests/test_views_search_as_you_type.py]: typo

File: invenio_records_rest/views.py [hunk 3]
```diff
@@ -182,7 +183,8 @@ def create_url_rules(endpoint, list_route=None, item_route=None,
                      default_media_type=None,
                      max_result_window=None, use_options_view=True,
                      search_factory_imp=None, links_factory_imp=None,
-                     suggesters=None, default_endpoint_prefix=None):
+                     suggesters=None, default_endpoint_prefix=None,
```
Comment (MJedr) [inline invenio_records_rest/views.py:186]: Maybe you can update the docstring?

File: invenio_records_rest/views.py (hunk 1 above)
Comment (drjova) [reply invenio_records_rest/views.py]: the `key` should always be `1`, what would you suggest for a name?

Comment (MJedr) [reply invenio_records_rest/views.py]: Anything different than key and indicating that what is that set, maybe `matched_search_as_you_type_fields`

File: invenio_records_rest/views.py [hunk 4]
```diff
@@ -941,6 +953,60 @@ def check_etag(self, etag, weak=False):
...
+        """Get results from `search_as_you_type`."""
+        request_fields = set(request.values.keys())
+        key = self.search_as_you_type_fields & request_fields
+
+        if not key:
+            raise SuggestNoCompletionsRESTError(
+                ', '.join(sorted(self.search_as_you_type.keys())))
+
+        key = key.pop()
+        value = request.values.get(key)
```
Comment (MJedr) [inline invenio_records_rest/views.py]: What if it would be None?

Comment (drjova) [reply invenio_records_rest/views.py]: It will be an empty string because the value it's coming from the request values

**Comment by drjova** [comment]
@MJedr updated

**Review by MJedr — APPROVED** [review]

**Pull request merged**